Named-exports extras - Default export function with named exports

Sauloxd opened the pull request:
Related issue: #2184 

This enhances the `extras/named-exports.js` to allow enumerable keys from default exported functions to be handled as named exports (previous behaviour was to select by object type).

I'm not sure if I should add an UMD bundle type fixture (as they seem kinda redundant with AMD fixtures), but it doest feels closer to the problem described in the issue. I can change that if it's better!

With the previous build, only the test regarding `named exports + default object export` passed (as it was the previous behaviour) and the `named exports + default function export` fails
With the new build, all tests passes.

### Obs
I have no idea why `dist/system-node.cjs` file has extra code included (I built it with: `yarn install` + `yarn build`). Maybe it's due to my node-version/yarn-version being different or something like that?

joeldenning commented on src/extras/named-exports.js:
```diff
@@ -42,10 +42,10 @@
             for (var exportName in defaultExport) {
               // default is not a named export
-              if (exportName !== 'default') {
+              if (defaultExport.hasOwnProperty(exportName) && exportName !== 'default') {
```
I'm wondering if there is ever a case where the defaultExport does not have a `hasOwnProperty` function. The `defaultExport` can have any type of value. I've checked and numbers, strings, symbols, functions, objects, and arrays and they do have it. However, it's possible that someone could modify the prototype of the value such that it doesn't have a default export. An example is that ES module "namespace" objects do not have the object prototype, and are created with `Object.create(null)`.

A tangential example where the object prototype is removed from some things - https://github.com/systemjs/systemjs/issues/2147. That particular example doesn't demonstrate a specific case where the code here would break things, but just shows that there are situations where values do not have the Object prototype.

In such cases where hasOwnProperty does not exist, this code will throw an error. Perhaps the safer implementation would be this?

```suggestion
              if (Object.prototype.hasOwnProperty.call(defaultExport, exportName) && exportName !== 'default') {
```

guybedford replied:
Yes, `Object.hasOwnProperty` is preferred for this reason.

joeldenning commented on test/fixtures/browser/umd-default-function-with-named-exports-module.js:
```diff
@@ -0,0 +1,12 @@
+  if (typeof define === 'function' && define.amd) {
+      define(['./amd-dep.js'], factory);
+  } else {
+    throw 'Test suit only supports amd'
```
In this project, we usually use semicolons:

```suggestion
    throw 'Test suit only supports amd';
```

joeldenning commented on test/fixtures/browser/umd-default-object-with-named-exports-module.js:
```diff
@@ -0,0 +1,13 @@
+  if (typeof define === 'function' && define.amd) {
+      define(['./amd-dep.js'], factory);
+  } else {
+    throw 'Test suit only supports amd'
```
```suggestion
    throw 'Test suit only supports amd';
```

Sauloxd commented:
Thanks for reviewing guys! I pushed a new version with your suggestions.

Updated:
- Use Object.protootype.hasOwnProperty instead of depending on export defaults always being inherited from Object
- Added a new test for export defaults originated by Object.create(null)
